src/sql/workbench/contrib/welcome/notifyEncryption/notifyEncryptionDialog.ts

```diff
@@ -33,19 +35,23 @@ export class NotifyEncryptionDialog extends ErrorMessageDialog {
 		@ITextResourcePropertiesService textResourcePropertiesService: ITextResourcePropertiesService,
 		@IOpenerService openerService: IOpenerService,
 		@IInstantiationService private _instantiationService: IInstantiationService,
-		@IStorageService private _storageService: IStorageService
+		@IStorageService private _storageService: IStorageService,
+		@IConnectionManagementService private _connectionManagementService: IConnectionManagementService,
 	) {
 		super(themeService, clipboardService, layoutService, telemetryService, contextKeyService, logService, textResourcePropertiesService, openerService);
 	}
 
 	public override open(): void {
-		if (this._storageService.get(NotifyEncryptionDialog.NOTIFY_ENCRYPT_SHOWN, StorageScope.GLOBAL)) {
+		if (this._storageService.get(NotifyEncryptionDialog.NOTIFY_ENCRYPT_SHOWN, StorageScope.GLOBAL)
+			|| !this._connectionManagementService.getConnections()?.some(conn => conn.providerName === mssqlProviderName)) {
+			// avoid connections added later triggering this dialog
+			this._storageService.store(NotifyEncryptionDialog.NOTIFY_ENCRYPT_SHOWN, true, StorageScope.GLOBAL, StorageTarget.MACHINE);
```

Comment:
otherwise a new user / a user with no existing connections will get this dialog if they add a new MSSQL connection later on.